querygen/querygen.go

```diff
@@ -7,29 +7,53 @@ package querygen
 func GenApplicationQuery(index string) string {
 	q, _ := UserJourneyQueryGen(
 		index,
-		`verb=create ` +
-		`"responseStatus.code" IN (200, 201) ` +
-		`"objectRef.apiGroup"="appstudio.redhat.com" ` +
-		`"objectRef.resource"="applications" ` +
-		`("impersonatedUser.username"="*" OR (user.username="*" AND NOT user.username="system:*")) ` +
-		`(verb!=create OR "responseObject.metadata.resourceVersion"="*")`,
+		`verb=create `+
+			`"responseStatus.code" IN (200, 201) `+
+			`"objectRef.apiGroup"="appstudio.redhat.com" `+
+			`"objectRef.resource"="applications" `+
+			`("impersonatedUser.username"="*" OR (user.username="*" AND NOT user.username="system:*")) `+
+			`(verb!=create OR "responseObject.metadata.resourceVersion"="*")`,
 		[]string{"name", "userId"},
 	)
 	return q
 }
 
-// GenPipelineRunQuery returns a Splunk query for generating Segment events
+// GenBuildPipelineRunCreatedQuery returns a Splunk query for generating Segment events
 // representing creation of AppStudio build PipelineRuns.
-func GenPipelineRunQuery(index string) string {
+func GenBuildPipelineRunCreatedQuery(index string) string {
 	q, _ := UserJourneyQueryGen(
 		index,
-		`verb=create ` +
-		`"responseStatus.code" IN (200, 201) ` +
-		`"objectRef.apiGroup"="tekton.dev" ` +
-		`"objectRef.resource"="pipelineruns" ` +
-		`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build` +
-		`"responseObject.metadata.resourceVersion"="*"`,
-		[]string{"namespace","application","component"},
+		`verb=create `+
+			`"responseStatus.code" IN (200, 201) `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="pipelineruns" `+
+			`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`| eval event_subject="Build PipelineRun" `,
+		[]string{"namespace", "application", "component"},
+	)
+	return q
+}
+
+// GenBuildPipelineRunStartedQuery returns a Splunk query for generating Segment events
+// representing the start of AppStudio build PipelineRuns.
+func GenBuildPipelineRunStartedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="pipelineruns" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.labels.pipelines.appstudio.openshift.io/type"=build `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.startTime"="*" `+
+			`| eval event_subject="Build PipelineRun", event_verb="started"`+
```

Comment:
might as well pass-through the complete `event` string instead of hacking subject and verb like this.

this will remove the need to modify the field definitions for them.

Reply:
Changed. I did have to create a field definition for `event` and some simple fallback logic in `splunk-to-segment.sh` to prioritize using this field over the combination of `event_subject` and `event_verb`.

Reply:
Didn't we have this already? I remember making an issue for it...

Reply:
Perhaps you're thinking of [RHTAPWATCH-373](https://issues.redhat.com/browse/RHTAPWATCH-373)? Some of the AC in [RHTAPWATCH-384](https://issues.redhat.com/browse/RHTAPWATCH-384) touches upon this behavior too.